ROADMAP.md

```diff
@@ -0,0 +1,116 @@
+-   (Google exploring) Build a Python SDK using a MicroPython port.
+-   (Help wanted) Stop using Emscripten in the C++ SDK. Instead use Clang /
+    wasi-sdk (proxy-wasm/proxy-wasm-cpp-sdk#167).
+-   (Help wanted) Build a Lua SDK using a Lua interpreter. Creates a safer
+    alternative to Envoy's
+    [Lua filter](https://www.envoyproxy.io/docs/envoy/latest/configuration/http/http_filters/lua_filter),
+    and could benefit Nginx's Lua-based [OpenResty](https://openresty.org/)
+    ecosystem.
+
+<a name="host"></a>
+
+## Host features
+
+-   (@mpwarres) CppHost maintenance.
+    -   Update v8 and upstream some patches for v8 warming / extension.
+    -   Update the protobuf dependency.
+    -   Set up dependabot.
+-   (Help wanted) Prototype
+    [HyperLight](https://github.com/hyperlight-dev/hyperlight-wasm) as a
+    KVM-based sandboxing layer around wasm runtimes. The allure is getting an
+    inexpensive and transparent second layer of security at a thread boundary,
+    which makes it more feasible to run fully untrusted workloads with
+    Proxy-Wasm.
+-   (Help wanted) Performance benchmarks. One of Proxy-Wasm's strengths is its
+    ability to swap between multiple wasm runtimes. Help users make an informed
+    decision by benchmarking cold start and execution costs across runtimes.
+-   (Help wanted) Adopt CPU metering as a first-class feature. Leverage
+    instruction counting where available. For other engines (e.g. v8), use a
+    watchdog thread.
+-   (Help wanted) Support dynamic (per VM) limits for RAM and CPU.
+-   (Help wanted) Expand the use of SharedArrayBuffer to reduce memcpy into wasm
+    runtimes. This is promising for HTTP body chunks (see relevant
+    [WASI issue](https://github.com/WebAssembly/WASI/issues/594)) and
+    [wasm binaries](https://github.com/proxy-wasm/proxy-wasm-cpp-host/blob/21a5b089f136712f74bfa03cde43ae8d82e066b6/src/v8/v8.cc#L272).
+-   (Help wanted) Implement NullVM for Rust and/or Go. For proxy owners with
+    trusted extensions, achieve native performance while maintaining
+    WebAssembly's portability.
+
+<a name="envoy"></a>
+
+## Envoy integration
+
+-   (@mpwarres, @botengyao) Get Envoy's inline wasm filter out of alpha
+    (envoyproxy/envoy#36996). Documentation, security scanning, tests, bug
+    fixes, etc.
+-   (@mpwarres) Implement the v0.3 Proxy-Wasm ABI.
+-   (Help wanted) Decouple from the thread-local execution model. As wasm
+    modules become more CPU intensive and leverage multiple async APIs, consider
+    managing a separate Proxy-Wasm threadpool. Each VM needs a work queue, and
+    requests need affinity to a single VM. This architecture allows for
+    independent thread scaling (expensive wasms get more CPU), improved
+    parallelism (multiple requests' wasm at the same time), and reduced memory
+    costs (one VM serves multiple Envoy threads). It adds performance risks (CPU
+    scheduling latency, CPU cache misses, NUMA hopping).
```

Comment:
drive-by comment - I would love this as an option, but I'm not sure I would love it as a one-size-fits-all due to the concerns mentioned. Thinking out loud; I wonder if the problem could be addressed by offering the decoupling as an api to modules instead, something along the lines of that modules could opt in to posting work on a threadpool (and register for completions or some such)?